src/Microsoft.ApiDesignGuidelines.Analyzers/Microsoft.ApiDesignGuidelines.Analyzers.md

```diff
@@ -122,6 +152,16 @@ Severity: Warning
 
 Help: [https://msdn.microsoft.com/library/ms182153.aspx](https://msdn.microsoft.com/library/ms182153.aspx)
 
+### CA1034: Nested types should not be visible ###
+
+A nested type is a type that is declared in the scope of another type. Nested types are useful to encapsulate private implementation details of the containing type. Used for this purpose, nested types should not be externally visible.
```

Comment:
The last sentence suggests that there are other purposes for which externally visible public types are appropriate? Is that true? If so, do we want to expand on that here?